DQM/EcalPreshowerMonitorModule/src/ESIntegrityTask.cc

```diff
@@ -67,9 +67,12 @@ void ESIntegrityTask::dqmEndRun(const Run& r, const EventSetup& c) {
   // TODO: no longer possible, clone histo beforehand if full statisticcs at end of run are required.
 }
 
-void ESIntegrityTask::dqmBeginLuminosityBlock(const edm::LuminosityBlock& lumi, const edm::EventSetup& c) {
+std::shared_ptr<ESLSCache> ESIntegrityTask::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
+                                                                       const edm::EventSetup& c) const {
   LogInfo("ESIntegrityTask") << "analyzed " << ievt_ << " events";
   // In case of Lumi based analysis SoftReset the Integrity histogram
+  auto lumiCache = std::make_shared<ESLSCache>();
+  lumiCache->ievtLS_ = 0;
```

Comment:
Below here you reset the MonitorElements. That behavior is incompatible with concurrent Lumis as you are sharing the same MonitorElement so you reset it while events of the previous Lumi are still being processed.

Reply:
@Dr15Jones In that case, is it impossible to have per-lumisection resets with concurrent processing? What if the MonitorElement were part of the luminosity cache (book a new histogram for each lumisection) -- would that work? Or is there another officially supported solution? Sorry, one reason this fix was difficult for ECAL was I tried to look through other subsystems' code and couldn't find these resets...